### PR TITLE
Fix room list display with empty notification actions

### LINKED ISSUE #181: Failure to list rooms after succesful login
Hi @alphapapa  

Thanks for your great work! I'm having some problems getting ement.el to work. My *Messages* buffer lists the following:

```
Ement: Response arrived after 10.08 seconds.  Reading 1.4k JSON response...
Ement: Reading JSON took 0.00 seconds
Ement: Reading events... 
Ement: Sync request sent, waiting for response...
Invalid face attribute :family nil [2 times]
error in process sentinel: ement-room-list-column-format-notifications: Args out of range: [], 0
```

I'm on Emacs version 29.0.91 and element version 0.10

Thanks!

### PR DESCRIPTION
**Pull request by Stebalien**
Parse actions according to https://spec.matrix.org/v1.7/client-server-api/#actions, allowing for 0 or more actions in any order.

fixes #181

### CODE REVIEW

File: ement-lib.el [hunk 1]
```diff
@@ -597,12 +597,12 @@ Returns one of nil (meaning default rules are used), `all-loud',
...
-                             (equal "dont_notify" (elt (alist-get 'actions rule) 0))))
+                 (rule) (when-let ((actions (alist-get 'actions rule)))
+                          (seq-contains-p actions "dont_notify")))
                 (tweak-rule-p
-                 (type rule) (pcase-let (((map ('actions `[,action ,alist])) rule))
-                               (and (equal "notify" action)
-                                    (equal type (alist-get 'set_tweak alist))))))
+                 (type rule) (when-let ((actions (alist-get 'actions rule)))
+                               (and (seq-contains-p actions "notify")
+                                    (seq-contains-p actions `(set_tweak . ,type) 'seq-contains-p)))))
```
Comment (Stebalien) [inline ement-lib.el:605]: I _think_ this last line is correct, but I'm not sure how to test it.

File: ement-lib.el [hunk 2]
```diff
@@ -597,12 +597,12 @@ Returns one of nil (meaning default rules are used), `all-loud',
                                          (equal "room_id" key)
                                          (equal (ement-room-id room) pattern)))))
                 (mute-rule-p
-                 (rule) (and (= 1 (length (alist-get 'actions rule)))
-                             (equal "dont_notify" (elt (alist-get 'actions rule) 0))))
+                 (rule) (when-let ((actions (alist-get 'actions rule)))
+                          (seq-contains-p actions "dont_notify")))
```
Comment (Stebalien) [inline ement-lib.el:601]: I'm preserving the current logic here but "dont_notify" should be ignored by spec. Instead, ement is supposed to notify iff the "notify" action is present.

See the note in https://spec.matrix.org/v1.7/client-server-api/#actions.

File: ement-lib.el [hunk 3]
```diff
@@ -597,12 +597,12 @@ Returns one of nil (meaning default rules are used), `all-loud',
...
                 (tweak-rule-p
-                 (type rule) (pcase-let (((map ('actions `[,action ,alist])) rule))
```
Comment (Stebalien) [inline ement-lib.el:603]: `actions` being empty was causing the referenced bug.

**Comment by alphapapa** [comment]
Thanks for working on this.  I've been busy lately and haven't had time to work on Ement stuff, but I hope to review and merge this soon.

**Comment by Stebalien** [comment]
Np, been there. Take your time.

**Comment by ifitzpat** [comment]
Thanks @Stebalien  I can confirm this fixes the issue!

**Comment by alphapapa** [comment]
This is probably the correct fix, but I don't fully understand the problem, e.g. why I'm not getting this behavior.  I wonder if it's related to the homeserver.  My account is on matrix.org; what about yours, @Stebalien and @ifitzpat ?

**Comment by Stebalien** [comment]
Matrix.org (through pantalaimon). But I _think_ the issue is simply notification settings. IIRC, this happened after I changed some settings in a recentish version of element.

**Comment by ifitzpat** [comment]
Also matrix.org

**Comment by alphapapa** [comment]
> Matrix.org (through pantalaimon). But I _think_ the issue is simply notification settings. IIRC, this happened after I changed some settings in a recentish version of element.

If there's any way you could figure out exactly what you changed the settings from and to, that might be helpful.

**Comment by Stebalien** [comment]
I really don't know. I didn't sign in through ement for a while before running into this issue.

**Pull request merged**